From ff0ef08925dbf0216ed4a40fcd1f360c30223061 Mon Sep 17 00:00:00 2001
From: Sergio Lopez <slp@redhat.com>
Date: Wed, 4 Sep 2024 14:30:37 +0200
Subject: [PATCH 3/3] vm_tools: sommelier: don't call to fixup_plane0

The function sl_linux_dmabuf_fixup_plane0_params relies on the
DRM_IOCTL_VIRTGPU_RESOURCE_INFO_CROS ioctl which is only available on
Chrome OS kernel + userspace.

Avoid calling it and assume our hw buffers are always virtgpu buffers.

Signed-off-by: Sergio Lopez <slp@redhat.com>
---
 compositor/sommelier-drm.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/compositor/sommelier-drm.cc b/compositor/sommelier-drm.cc
index 1ed50394e9..67a6a35f15 100644
--- a/compositor/sommelier-drm.cc
+++ b/compositor/sommelier-drm.cc
@@ -98,8 +98,7 @@ static void sl_drm_create_prime_buffer(struct wl_client* client,
   uint32_t host_modifier_lo = DRM_FORMAT_MOD_INVALID & 0xFFFFFFFF;
   bool is_virtgpu_buffer = false;
   if (ctx->gbm) {
-    is_virtgpu_buffer = sl_linux_dmabuf_fixup_plane0_params(
-        ctx->gbm, name, &host_stride0, &host_modifier_hi, &host_modifier_lo);
+    is_virtgpu_buffer = true;
   }
 
   zwp_linux_buffer_params_v1_add(buffer_params, name, 0, offset0, host_stride0,
-- 
2.45.1

